dumpling, lightning: add document for compressed files support (#11418)
pingcap/docs: tidb-lightning/tidb-lightning-data-source.md

For more information on `mydumper.files`, refer to [Match customized file](#match-customized-files).

## Compressed files

TiDB Lightning currently supports compressed files exported by Dumpling or compressed files that follow the naming rules. Currently, TiDB Lightning supports the following compression algorithms: `gzip`, `snappy`, and `zstd`. When the file name follows the naming rules, TiDB Lightning automatically identifies the compression algorithm and imports the file after streaming decompression, without additional configuration.

> **Note:**
>

> > - Because TiDB Lightning cannot concurrently decompress a single large compressed file, the size of the compressed file directly affects the import speed. It is recommended that the size of the source file after decompressing does not exceed 256 MiB.

Reviewer: Not so sure
```suggestion
> - Because TiDB Lightning cannot concurrently decompress a single large compressed file, the size of the compressed file directly will affect the import speed. It is recommended that the size of the source file after decompressing does not exceed 256 MiB.
```

> - TiDB Lightning only supports importing data files that are compressed separately. It does not support importing a collection of compressed data files that are compressed together.

> > - TiDB Lightning does not support `parquet` files compressed through another compress tool, such as `db.table.parquet.snappy`. If you want to compress `parquet` files, you can configure the compression format for the `parquet` file writer.

Author: ```suggestion
> - TiDB Lightning does not support `parquet` files compressed through another compression tool, such as `db.table.parquet.snappy`. If you want to compress `parquet` files, you can configure the compression format for the `parquet` file writer.
```

> > - TiDB Lightning does not support `parquet` files compressed with a second compression algorithm, such as `db.table.parquet.snappy`. If you want to compress `parquet` files, you can configure the compression format for the `parquet` file storage.

Reviewer: ```suggestion
> - TiDB Lightning does not support `parquet` files compressed through another compress tool, such as `db.table.parquet.snappy`. If you want to compress `parquet` files, you can configure the compression format for the `parquet` file writer.
```

> > - TiDB Lightning v6.4 and later versions do not support data files with a non-`.bak` suffix and report errors. You need to modify the file name in advance, or move the file out of the import data directory to avoid such errors.

Reviewer: ```suggestion
> - TiDB Lightning v6.4 and later versions do not support data files with a non-`.bak` suffix and report errors. You need to modify the file name in advance, or move these files out of the import data directory to avoid such errors.
```

> > - TiDB Lightning v6.4 and later versions do not support data files with a non-`.bak` suffix and report errors. You need to modify the file name in advance, or move these files out of the import data directory to avoid such errors.

Author: ```suggestion
> - TiDB Lightning v6.4 and later versions only support data files with the following suffixes: `gzip`, `snappy`, `zstd` and `.bak`. Other suffixes cause errors. For a file with unsupported suffix, you need to modify the file name in advance, or move the file out of the import data directory to avoid such errors.
```

> > - TiDB Lightning v6.4 and later versions only support data files with the following suffixes: `gzip`, `snappy`, `zstd` and `.bak`. Other suffixes cause errors. For a file with unsupported suffix, you need to modify the file name in advance, or move the file out of the import data directory to avoid such errors.

Author: ```suggestion
> - TiDB Lightning v6.4 and later versions only support `.bak` files and the following compressed data files: `gzip`, `snappy`, and `zstd`. Other types of files cause errors. For those unsupported files, you need to modify the file names in advance, or move those files out of the import data directory to avoid such errors.
```

> > - TiDB Lightning v6.4 and later versions only support `.bak` files and the following compressed data files: `gzip`, `snappy`, and `zstd`. Other types of files cause errors. For those unsupported files, you need to modify the file names in advance, or move those files out of the import data directory to avoid such errors.

Reviewer 2: ```suggestion
> - TiDB Lightning v6.4.0 and later versions only support `.bak` files and the following compressed data files: `gzip`, `snappy`, and `zstd`. Other types of files cause errors. For those unsupported files, you need to modify the file names in advance, or move those files out of the import data directory to avoid such errors.
```
